### PR TITLE
##3760702 get roles of a user and users with a particular role

### PR DESCRIPTION
**Pull request by andela-ojoloko**
#### What does this PR do?
Implement get all roles for single user and get all users that have a particular role.
#### Description of Task to be completed?
- Create role model
- Update Admin access Decorator
- Create endpoint `get api/v1/users/:user_id/roles` to get a all roles for a particular user
- Create endpoint `get api/v1/roles/:role_id/users` to get a all users that have a particular role.
- Write tests for endpoints
#### How should this be manually tested?
`pytest -V` or `pytest`
#### What are the relevant pivotal tracker stories?
151844008, 151843683, 151844209

### CODE REVIEW

File: test/api/test_models.py [hunk 1]
```diff
@@ -424,3 +426,80 @@ def test_pagination(self):
...
+        self.assertEqual(saved_role.name, self.role2.name)
+    
+    def test_add_role_user(self):
+        """
+        Tests succesfull adding of a user to role
+        """
+        self.role2.save()
+        self.role3.save()
+        self.role1.users.append(self.user1)
+        saved_role1 = Role.query.get(self.role1.id)
```
Comment (wanjiku-carol) [inline test/api/test_models.py]: Class method  here as well`Role.get(self.role1.id)`

File: test/api/views/test_roles.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            'email': 'test.test@andela.com',
+            'first_name': 'test',
+            'last_name': 'test',
+            'name': 'test test',
+            'picture': 'https://lh6.googleusercontent.com/-1DhBLOJentg'
+            '/AAAAAAAAAAI/AAAAAAAAABc/ImM13eP_cAI/photo.jpg?sz=50',
+            'roles': {
+                    'Andelan': '-Ktestandelanid',
+                    'Fellow': '-Ktestfellowid'
+            }
```
Comment (wanjiku-carol) [inline test/api/views/test_roles.py]: Is line 20 to 23 still relevant?

File: test/api/views/test_roles.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            "email": "testadmin.test@andela.com",
+            "first_name": "admin",
+            "last_name": "admin",
+            "name": "admin admin",
+            "picture": "https://lh6.googleusercontent.com/-1DhBLOJentg"
+                       "/AAAAAAAAAAI/AAAAAAAAABc/ImM13eP_cAI/photo.jpg?sz=50",
+            "roles": {
+                "Andelan": "-Ktestandelanid",
+                "Fellow": "-Ktestfellowid"
+            }
```
Comment (wanjiku-carol) [inline test/api/views/test_roles.py]: Is line 39 to 42 still relevant?

File: test/api/views/test_users.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            "last_name": "admin",
+            "name": "admin admin",
+            "picture": "https://lh6.googleusercontent.com/-1DhBLOJentg"
+                       "/AAAAAAAAAAI/AAAAAAAAABc/ImM13eP_cAI/photo.jpg?sz=50",
+            "roles": {
+                "Andelan": "-Ktestandelanid",
+                "Fellow": "-Ktestfellowid",
+            }
+        },
+        "exp": exp_date + datetime.timedelta(days=1)
```
Comment (wanjiku-carol) [inline test/api/views/test_users.py]: Is line 35 - 38 still relevant?

File: test/api/views/test_roles.py (hunk 2 above)
Comment (andela-ojoloko) [reply test/api/views/test_roles.py]: No. It is an example of what the token that comes from the Andelan microservice contains. I believe it is better we keep it that way.

File: api/views/users.py [hunk 5]
```diff
@@ -34,3 +34,25 @@ def post(self):
...
             response.status_code = 400
         return response
+
+
+class UserRoleResource(Resource):
+
+    @token_required
+    def get(self, user_id):
+        user = User.query.get(user_id)
+
```
[the remark this answers is not in the document]
Comment (andela-ojoloko) [reply api/views/users.py]: Fixed

File: api/views/roles.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class RoleUsersResource(Resource):
+    """Contain GET users with a particular role."""
+
+    @token_required
+    def get(self, role_id):
+        """Return list of all users with a particular role."""
+        try:
+            users = (Role.query.get(role_id)).users.all()
```
[the remark this answers is not in the document]
Comment (andela-ojoloko) [reply api/views/roles.py]: Fixed

File: test/api/views/test_roles.py (hunk 3 above)
Comment (andela-ojoloko) [reply test/api/views/test_roles.py]: No. It is a replica of what the token that comes from the Andelan microservice contains. I believe it is better we keep it that way.

**Comment by wanjiku-carol** [comment]
LGTM

**Comment by Mnickii** [comment]
LGTM

**Pull request merged**